spec/rubocop/cop/style/empty_lines_around_exception_handling_keywords_spec.rb

```diff
@@ -0,0 +1,235 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Style::EmptyLinesAroundExceptionHandlingKeywords do
+  let(:config) { RuboCop::Config.new }
+  subject(:cop) { described_class.new(config) }
+
+  shared_examples :offense do |name, code, correction|
+    it "registers an offense for #{name} with a blank" do
+      inspect_source(cop, code)
+      expect(cop.offenses.size).to eq(1)
+    end
+
+    it "autocorrects for #{name} with a blank" do
+      corrected = autocorrect_source(cop, code)
+      expect(corrected).to eq(correction)
+    end
+  end
+
+  shared_examples :accepts do |name, code|
+    it "accepts #{name}" do
+      inspect_source(cop, code)
+      expect(cop.offenses).to be_empty
+    end
+  end
+
+  include_examples :offense, 'above rescue keyword', <<-CODE, <<-CORRECTION
+begin
+  f1
+
+rescue
+  f2
+end
+  CODE
+begin
+  f1
+rescue
+  f2
+end
+  CORRECTION
+  include_examples :offense, 'rescue section starting', <<-CODE, <<-CORRECTION
+begin
+  f1
+rescue
+
+  f2
+end
+  CODE
+begin
+  f1
+rescue
+  f2
+end
+  CORRECTION
+  include_examples :offense, 'rescue section ending', <<-CODE, <<-CORRECTION
+begin
+  f1
+rescue
+  f2
+
+else
+  f3
+end
+  CODE
+begin
+  f1
+rescue
+  f2
+else
+  f3
+end
+  CORRECTION
+  include_examples :offense,
+                   'rescue section ending for method definition',
+                   <<-CODE, <<-CORRECTION
+def foo
```

Comment:
I really hate outdented hereblocks. Use the `strip_margin` method from `powerpack` to pretty those up a bit.